app/models/loader/orchestrate.rb

```diff
@@ -255,13 +255,67 @@ def insert_new
       @new_roles = ::Role.all
 
       in_primary_schema do
-        TABLES.each do |table|
-          columns = (TABLE_EQUIVALENCE_COLUMNS[table] + [ :policy_id ]).join(", ")
-          db.execute "INSERT INTO #{table} ( #{columns} ) SELECT #{columns} FROM #{schema_name}.#{table}"
-        end
+        disable_policy_log_trigger
+        TABLES.each { |table| insert_table_records(table) }
+        enable_policy_log_trigger
       end
     end
 
+    def insert_table_records(table)
+      columns = (TABLE_EQUIVALENCE_COLUMNS[table] + [ :policy_id ]).join(", ")          
+      db.run "INSERT INTO #{table} ( #{columns} ) SELECT #{columns} FROM #{schema_name}.#{table}"
+      
+      # For large policies, the policy logging triggers occupy the majority
+      # of the policy load time. To make this more efficient on the initial
+      # load, we disable the triggers and update the policy log in bulk.
+      insert_policy_log_records(table)
+    end
+
+    def disable_policy_log_trigger
+      # To disable the triggers during the bulk load, I tried a couple of
+      # methods before landing on the current:
+      #
+      # 1. `ALTER TABLE table ENABLE/DISABLE TRIGGER {trigger name}`
+      #     This didn't work because `ALTER TABLE` obtains an `ACCESS 
+      #     EXCLUSIVE LOCK`, which breaks concurrent policy loading.
+      #
+      # 2. `SET session_replication_role = replica;`
+      #     This didn't work because it disables *ALL* triggers for the
+      #     current session, and there are a number of other triggers that 
+      #     are necessary, and would become cumbersome to re-implement as bulk ops.
+      #
+      # 3. Configuration setting
+      #     The method I settled on was to use a configuration setting, scoped
+      #     to the session, that the trigger function is aware of. When we set
+      #     this setting to bypass the trigger to `true`, then the trigger will
+      #     observe the setting value and not create its own policy log.
+      db.run 'SET myvars.skip_insert_policy_log_trigger = true'
+    end
+
+    def enable_policy_log_trigger
+      db.run 'SET myvars.skip_insert_policy_log_trigger = false'
+    end
+
+    def insert_policy_log_records(table)
+      primary_key_columns = Array(Sequel::Model(table).primary_key).map(&:to_s).pg_array
+      db.run <<-POLICY_LOG
+          INSERT INTO policy_log(
```

Comment:
I think it's important that we ensure that policy log entries generated here match those generated when `myvars.skip_insert_policy_log_trigger` is `true`.

Seems like there are two options: 

* DRY up the code, and have both paths call a function to do the `INSERT INTO ... SELECT`.
* Add an automated test that shows that entries generated both ways are identical.

I think the first option is preferable, but it's not immediately obvious to me that it's possible. @micahlee, can you investigate and see what you can come up with?